src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ForbidPrivateEventHandlers/ForbidPrivateEventHandlersAnalyzer.cs

```diff
@@ -0,0 +1,129 @@
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ForbidPrivateEventHandlers
+{
+	public class ForbidPrivateEventHandlersAnalyzer : PXGraphAggregatedAnalyzerBase
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldNotBePrivate,
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations);
+		
+		private static readonly Dictionary<SyntaxKind, string> ModifierText = new()
+		{
+			{ SyntaxKind.PublicKeyword, "public" },
+			{ SyntaxKind.ProtectedKeyword, "protected" },
+			{ SyntaxKind.VirtualKeyword, "virtual" }
+		};
+
+		private static readonly Dictionary<Accessibility, SyntaxKind> AccessibilitySyntaxKindMap = new()
+		{
+			{ Accessibility.Public, SyntaxKind.PublicKeyword },
+			{ Accessibility.Protected, SyntaxKind.ProtectedKeyword }
+		};
+
+		public override void Analyze(SymbolAnalysisContext context, PXContext pxContext, PXGraphEventSemanticModel graphOrGraphExtension)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+			
+			var declaredEventHandlers = graphOrGraphExtension
+				.GetAllEvents()
+				.Where(graphEvent => graphEvent.Symbol.IsDeclaredInType(graphOrGraphExtension.Symbol));
+
+			var allInterfaceMethods = graphOrGraphExtension
+				.Symbol
+				.AllInterfaces
+				.SelectMany(im => im.GetMethods())
+				.Where(im => im.GetEventHandlerType(pxContext) != EventType.None)
+				.ToList();
+
+			foreach (var handler in declaredEventHandlers)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var location = handler.Symbol.Locations.FirstOrDefault();
+
+				if (location == null)
+				{
+					continue;
+				}
+
+				if (handler.Symbol.IsOverride)
+				{
+					continue;
+				}
+
+				var properties = new Dictionary<string, string?>
+				{
+					{ DiagnosticProperty.IsContainingTypeSealed, handler.Symbol.ContainingType.IsSealed.ToString() }
+				};
+
+				if (handler.Symbol.MethodKind == MethodKind.ExplicitInterfaceImplementation)
+				{
+					context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations, location, properties));
+				}
+				else if (!IsImplicitInterfaceImplementation(handler.Symbol, allInterfaceMethods))
+				{
+					var targetAccessibility = handler.Symbol.ContainingType.IsSealed ? Accessibility.Public : Accessibility.Protected;
+					var addVirtualModifier = !handler.Symbol.ContainingType.IsSealed && !handler.Symbol.IsAbstract;
+
+					properties.Add(DiagnosticProperty.AddVirtualModifier, addVirtualModifier.ToString());
+					properties.Add(StaticAnalysis.DiagnosticProperty.RegisterCodeFix, true.ToString());
+
+					if (!(handler.Symbol.IsVirtual || handler.Symbol.ContainingType.IsSealed || handler.Symbol.IsAbstract) || handler.Symbol.DeclaredAccessibility != targetAccessibility)
```

Comment:
> `handler.Symbol.ContainingType.IsSealed || handler.Symbol.IsAbstract`

`addVirtualModifier` should be reused here. This will be more readable and guarantee that this check and `addVirtualModifier` calculation will be always the same.

Reply:
Done.